lib/src/io.dart

```diff
@@ -1072,12 +1072,14 @@ Future<void> extractTarGz(Stream<List<int>> stream, String destination) async {
       throw FormatException('Tar file contained duplicate path ${entry.name}');
     }
 
-    if (!p.isWithin(destination, filePath)) {
+    if (!(p.isWithin(destination, filePath) ||
+        // allow including '.' as an entry in the tar.gz archive.
+        p.equals(destination, filePath))) {
```

Comment:
Perhaps we should only allow this if the entry is also a directory, I think it might break for a regular file named `.` in the archive (since we then have a single file where we'd expect a directory).

Reply:
Good point! I think it would be caught below when trying to create the file, but it might be better to catch early.

Updated.